Runtime/JSArray.cs

```diff
@@ -0,0 +1,41 @@
+using System.Collections;
+using System.Collections.Generic;
+
+namespace NodeApi;
+
+public struct JSArray : IEnumerable<JSValue>, IEnumerable
+{
+    private JSValue _value;
+
+    public static explicit operator JSArray(JSValue value) => new() { _value = value };
+    public static implicit operator JSValue(JSArray arr) => arr._value;
+
+    public static explicit operator JSArray(JSObject obj) => (JSArray)(JSValue)obj;
+    public static implicit operator JSObject(JSArray arr) => (JSObject)arr._value;
+
+    public JSArray()
+    {
+        _value = JSValue.CreateArray();
+    }
+
+    public JSArray(int length)
+    {
+        _value = JSValue.CreateArray(length);
+    }
+
+    public int Length => _value.GetArrayLength();
+
+    public JSValue this[int index]
+    {
+        get => _value.GetElement(index);
+        set => _value.SetElement(index, value);
+    }
+
+    public void Add(JSValue item) => _value["push"].Call(_value, item);
+
+    public JSArrayItemEnumerator GetEnumerator() => new(_value);
```

Comment:
I think `JSArrayItemEnumerator` can be internal, or even a private nested class.

Reply:
Also I'd suggest naming it `JSArrayEnumerator`.

Reply:
The main goal here was to mimic the [`List.Enumerator`](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.enumerator?view=net-7.0) struct that allows enumerate `List` without extra memory allocations. I can make it a nexted class, but it must stay public to avoid memory allocations.

Reply:
Ahh that makes sense. I would prefer a nested `JSArray.Enumerator` class then, to follow the same pattern as `List` and avoid polluting the namespace.